great_tables/_boxhead.py

```diff
@@ -99,7 +100,18 @@ def cols_label(self: GTSelf, **kwargs: str) -> GTSelf:
     # msg: "All column names provided must exist in the input `.data` table."
     _assert_list_is_subset(mod_columns, set_list=column_names)
 
-    boxhead = self._boxhead._set_column_labels(kwargs)
+    # Handle units syntax in labels (e.g., "Density ({{ppl / mi^2}})")
+    new_kwargs: dict[str, UnitStr | str] = {}
+
+    for k, v in kwargs.items():
+        # If the text is a string with has units notation within (detectable by "{{" and "}}")
+        # convert it to a UnitStr object
+        if isinstance(v, str) and ("{{" in v and "}}" in v):
```

Comment:
Can't a UnitStr be created from any string? Is it necessary to check for "{{"? This will also flag the case `}} {{` which is not a unit pattern.

Reply:
From pairing, we changed this to use logic from `UnitStr`.